### ISSUE TITLE
Missing options in Terraform provider, else documentation is hard to follow if they are there

### ISSUE THREAD

**Issue by martyav** [rancher/rancher-docs#831]
SURE-6515

Pertains to https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/rke2-cluster-configuration#machineglobalconfig and possibly other pages

Docs are out of date/misleading and missing information.

> We should link the docs of machine_global_config options to the server and agent cli args and outline syntax for useage:
https://docs.rke2.io/reference/server_config
https://docs.rke2.io/reference/linux_agent_config

**Comment by martyav** [comment]
I had a second look at the ticket and am unsure if this is something for us or for the maintainers of the Terraform docs site, as many of the comments link to https://registry.terraform.io/providers/rancher/rancher2/latest/docs.

**Comment by martyav** [comment]
Per discussions with @Tejeev, we're going to draft a quickstart for using Terraform based on the scripts here: https://github.com/rancher/quickstart/tree/master

**Comment by Tejeev** [comment]
While Marty works on that, I'm working to get some clear answers from engineering for improvements to the descriptions and syntax examples for the items on our internal customer ticket so we can improve those aspects of the existing documentation

**Comment by martyav** [comment]
On Friday, it was suggested that we could create a page similar to this, but for the Terraform provider:

https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/rke2-cluster-configuration#cluster-config-file-reference

We would need to know what the minimum arguments are to use TF to deploy a cluster, and offer better descriptions/syntax than the TF docs.

**Comment by btat** [comment]
Parent issue resolved.

**Comment by Tejeev** [comment]
The documentation has been updated with requested examples with just-released TFP 4.0.0, see https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/cluster_v2 (PR with the changes - https://github.com/rancher/terraform-provider-rancher2/pull/1300)
